xarray/tests/test_combine.py

```diff
@@ -667,6 +667,28 @@ def test_combine_by_coords(self):
         def test_empty_input(self):
             assert_identical(Dataset(), combine_by_coords([]))
 
+    def test_combine_coords_unnamed_arrays(self):
+        objs = [
+            DataArray([0, 1], dims=("x"), coords=({"x": [0, 1]})),
+            DataArray([2, 3], dims=("x"), coords=({"x": [2, 3]})),
+        ]
+        expected = Dataset({"_": ("x", [0, 1, 2, 3])}, coords={"x": [0, 1, 2, 3]})
```

Comment:
We should return a DataArray in this case.

Reply:
So the idea is to start with a nested list of DataArrays and end up with a DataArray. This will change the return type from Dataset to (DataArray or Dataset).

Reply:
I wonder if `combine_by_coords` should accept a list of both DataArrays and Datasets? In such a case, I believe it should consider the DataArrays as a separate variable to those in the Datasets. The dummy variable would then be merged with the other variables in the resulting Dataset. Otherwise, this function could take either a list of DataArrays or a list of Datasets.

As for `combine_nested`, it doesn't make sense in my mind to have it take both DataArrays and Datasets at the same time, since it aligns its input into a hypercube according to the nested list structure.

I will have `combine_by_coords` take the "either or" approach for now. Let me know if you think it should take both types of input simultaneously.

Reply:
@aijams the case of a list of both DataArrays and Datasets is the same as@dcherian's case (3) [in this comment](https://github.com/pydata/xarray/issues/3248#issuecomment-526263925). We have to pass the objects down to merge so that it can raise the correct error.

I think it would be nice if we covered the case {DataArrays}->DataArray, but I'm not sure what the easiest way to do that is. If the dataarrays are unnamed, then the combined result will have some sort of default name, which we could check for and then demote to DataArray before returning. But if the DataArrays are named then at the return step I don't think we will know whether the input was originally a DA or DS before it got promoted?

I'm not sure